src/test/java/com/zilen/weather/service/WeatherServiceTest.java

```diff
@@ -26,7 +30,7 @@
     public void findByCityName() {
         WeatherDTO weather = new WeatherDTO();
         Mockito
-                .when(restTemplate.getForEntity("https://api.openweathermap.org/data/2.5/weather?q=Moscow&units=metric&appid=50b4a9bfff0df818b269d4a468128442", WeatherDTO.class))
+                .when(restTemplate.getForEntity(url + "Moscow&units=metric&appid=" + appid, WeatherDTO.class))
```

Comment:
on every mocked call, you need to verify that this call happened. You can read about verify here: https://www.baeldung.com/mockito-verify